File: 2019W22/preppindataw22.R
library(dplyr)
library(readr)
library(zoo)        

final <- read_csv("E:/PD Wk 22 Input.csv", 
                  col_types = cols(Date = col_date(format = "%d/%m/%Y"))) %>%
  mutate('Moving Avg Sales'=rollapply(`Sales`,7,mean,align='right',fill=NA)) %>%
  select(-'Sales')

View(final)

File: 2019/2019W22/PreppindataW22.py
import pandas as pd
from datetime import datetime

sales = pd.read_csv("E:\PD Wk 22 Input.csv")
sales['Date']=sales['Date'].apply(lambda x: datetime.strptime(x, "%d/%m/%Y").date())
sales['Moving Avg Sales'] = sales.rolling(7)['Sales'].mean()
sales.drop(['Sales'],axis=1,inplace = True)
